### PR TITLE
Tc/schedule form

### PR DESCRIPTION
**Pull request by CedThen**

### CODE REVIEW

**Comment by CedThen** [comment]
It looks a little bit rough but it does what its supposed to!

**Comment by showduhtung** [comment]
hmm so you're actually missing a few things, validations for each form input

File: src/components/Main/Schedule/NewScheduleForm.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            }
+          >
+            <MenuItem value={0}>Church Council</MenuItem>
+            <MenuItem value={1}>RE</MenuItem>
+          </Select>
+          <FormHelperText>Team</FormHelperText>
+          <Tooltip id="Team" text="Select who is able to edit this schedule" />
+        </FormControl>
+      </form>
+      <button onClick={() => onSubmit(scheduleTitle, startDate, endDate, view, team)}>
```
Comment (showduhtung) [inline src/components/Main/Schedule/NewScheduleForm.tsx:109]: this function should only fire if all validations are met.
1) Title shouldn't be longer than 32 characters
2) all fields must be filled and not null/empty

**Review by showduhtung — CHANGES_REQUESTED** [review]
once you make the changes to validation, you have my approval and you can go ahead and merge

File: src/components/Main/Schedule/Home.tsx [hunk 2]
```diff
@@ -1,51 +1,87 @@
...
 import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
 import { buttonTheme } from '../../../shared/styles/theme.js';
 import { useSpinner } from '../../../shared/styles/loading-spinner';
 
+import AddIcon from '@material-ui/icons/Add';
+
 export const Home = () => {
   const classes = useStyles();
-  useSpinner();
+  // useSpinner();
```
Comment (hchiam) [inline src/components/Main/Schedule/Home.tsx:23]: @CedThen already showed me a problem with useSpinner not playing nice with react query useMutation. Looks like we'll need a more involved solution for the spinner. I'll look into maybe an interface/wrapper to make it easy to specify when to start/stop spinner within API calls.

File: src/components/Main/Schedule/Home.tsx [hunk 3]
```diff
@@ -1,51 +1,87 @@
...
-      staleTime: 100000000000000,
+      staleTime: 100000000000000, //1157407.4 days fyi l o l
     },
   );
+  const [mutateAddSchedule] = useMutation(addSchedule, {
+    onSuccess: () => cache.invalidateQueries('schedulesData'), //causes the roleData query to call and update on success
+  });
 
-  if (status === 'loading') return <div>loading...</div>; // loading state
+  // if (status === 'loading') return <div>loading...</div>; // loading state
```
Comment (hchiam) [inline src/components/Main/Schedule/Home.tsx:43]: i think this can be deleted altogether - this never gets run when i tried it out for some reason

**Review by hchiam — COMMENTED** [review]
Actually, I ran your branch `tc/schedule-form` just now and the spinner seems to work fine on my machine. 
[EDIT: @CedThen - I got a fix for the spinner [here](https://github.com/House-on-a-Rock/tjc-scheduling-web/pull/36/files#diff-7b4f7e8958ec472ac00c2ace4e37d1e0f6b4508eb484c2d006e73be0e52a3695)]

File: src/components/Main/Schedule/Home.tsx (hunk 2 above)
Comment (hchiam) [reply src/components/Main/Schedule/Home.tsx:23]: @CedThen - I got a fix for the spinner [here](https://github.com/House-on-a-Rock/tjc-scheduling-web/pull/36/files#diff-7b4f7e8958ec472ac00c2ace4e37d1e0f6b4508eb484c2d006e73be0e52a3695)

**Pull request merged**